orchestrator/utils/get_subscription_dict.py

```diff
@@ -0,0 +1,17 @@
+from uuid import UUID
+
+from orchestrator.domain.base import SubscriptionModel
+from orchestrator.services.subscriptions import _generate_etag, build_extended_domain_model
+from orchestrator.utils.redis import from_redis, to_redis
+
+
+async def get_subscription_dict(subscription_id: UUID) -> tuple[dict, str]:
+    if cached_model := from_redis(subscription_id):
+        return cached_model  # type: ignore
+
+    subscription_model = SubscriptionModel.from_subscription(subscription_id)
+    subscription = build_extended_domain_model(subscription_model)
+    etag = to_redis(subscription)
+    if not etag:
+        etag = _generate_etag(subscription)
```

Comment:
I would say just rely on `to_redis` to do the right thing, as it checks if domain model caching is enabled

```suggestion
    etag = to_redis(subscription)
```

Reply:
but the domain model endpoint needs a etag currently, so thought doing it here is the best place to always have an etag

Reply:
Ah wait. `to_redis()` is actually updating the cache! This is not meant to be called on the fly, this is only done as part of workflows

Reply:
do we not want to call it after having it fetched once, when it isn't in the cache yet?
mostly a dev problem with importing DB's

Reply:
The step

```python
@step("Cache Subscription and related subscriptions")
def cache_domain_models(workflow_name: str, subscription: SubscriptionModel | None = None) -> State:
```

Is responsible for updating a subscription in the cache. This is called at the end of workflows.

If a subscription is not found in the cache, this means that something is wrong with it and that the workflow is stuck. When that happens, we should not have some other part of the code secretly fill the cache

Reply:
oke, removed it from `get_subscription_dict`